### ISSUE TITLE
Timeline mixup when running bagfile

### ISSUE THREAD

**Issue by glpuga** [Ekumen-OS/beluga#232]
### Bug description

A few transforms and messages are getting stamped with unmatched timestamps, probably as a result of a mixup when the bagfile was generated.

As a result of this, amcl is publishing the map->odom tf in a totally different timescale than the odom->base and base->laser_front , causing RViz to fail to display the later.

See timestamps on the logs and on the two transforms on the right:

![Screenshot from 2023-06-19 13-59-14](https://github.com/Ekumen-OS/beluga/assets/17802342/99f79579-f463-4076-b0a9-97330e0333e8)

See timestamps on the /clock and /scan topics:

![Screenshot from 2023-06-19 14-01-11](https://github.com/Ekumen-OS/beluga/assets/17802342/329910c4-487c-4238-be19-223d954ccb42)

On RViz,

![Screenshot from 2023-06-19 13-52-31](https://github.com/Ekumen-OS/beluga/assets/17802342/0d23acb9-84d1-4d06-a2ed-3616629117b6)

**Platform (please complete the following information):**
 - OS: Humble development container included in Beluga
 - `Beluga` version: ed7dfe38bcab566a900b636378e7dbf05981120a

### How to reproduce
_List steps to reproduce the issue:_
1. On a console: `cd /ws && colcon build && source install/setup.bash && ros2 launch beluga_example perfect_odometry.launch.xml`
2. On another console: `ros2 topic echo /tf`, `ros2 topic echo /clock`, `ros2 topic echo /scan`
3. On RViz, the TF visor will also show a warning for both base and laser_front.

_Code snippets or minimal examples are always helpful, if not necessary._

**Expected behavior**

All message timestamps to be consistent when running out of a demo and test bagfile.

**Actual behavior**

Timestamps are not consistent with a single timeline, and some information fails to be displayed as a result.

**Comment by hidmic** [comment]
I think I saw this when working on #223. Messages carry some simulated clock stamp but `rosbag2` stamped them on arrival using the system clock.

**Comment by glpuga** [comment]
This is probably related to `beluga_benchmark` being unable to measure NAV2 AMCL. The node crashes, regardless of the configuration.

Bisecting with git, the problem with AMCL seems to have started starting in #211

**Comment by nahueespinosa** [comment]
Ah, that was me.

I did two things that might have caused this:

1. I removed the `/clock` topic messages from the ROS bag.
2. I added the `--clock` option to the ROS bag play command.

The motivation was being able to play ROS bags that don't have `/clock` messages, like the ones we recorded in HQ (_do you have any thoughts on how to do this correctly?_).

Seems like it was not the right approach, we can definitely revert the changes to the ROS bag.
